components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2Service.java

```diff
@@ -647,10 +647,9 @@ public String getOauthApplicationState(String consumerKey) {
             OAuthAppDO appDO = OAuth2Util.getAppInformationByClientId(consumerKey);
             return appDO.getState();
         } catch (IdentityOAuth2Exception | InvalidOAuthClientException e) {
-            String msg = "Error while finding application state for application with client_id: " + consumerKey;
-            log.error(msg);
             if (log.isDebugEnabled()) {
-                log.debug(msg, e);
+                log.debug("Error while finding application state for application with client_id: " +
```

Comment:
This could only be considered a client error in case its an `InvalidOAuthClientException`. It should be handled in the original manner in case its an `IdentityOAuth2Exception`. WDYT?

Reply:
Fixed https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1745/commits/46410198e6640c719c0a7347ef3003a141e893cb